src/lib.rs

```diff
@@ -244,6 +241,16 @@ impl<F: PrimeField, PC: PolynomialCommitment<F>, D: Digest> Marlin<F, PC, D> {
             second_comms.iter().map(|p| p.commitment().clone()).collect(),
             third_comms.iter().map(|p| p.commitment().clone()).collect(),
         ];
+        let labeled_comms: Vec<_> = index_pk
+            .index_vk
+            .iter()
+            .cloned()
+            .zip(&AHPForR1CS::<F>::INDEXER_POLYNOMIALS)
+            .map(|(c, l)| LabeledCommitment::new(l.to_string(), c, None))
+            .chain(first_comms.iter().cloned())
+            .chain(second_comms.iter().cloned())
+            .chain(third_comms.iter().cloned())
+            .collect();
```

Comment:
Could you explain why this is needed? Why did Marlin pre-fork not commit to the labels during fiat shamir as well? cc @Pratyush

Reply:
Do we plug labeled_comms into FS?